src/Pact/Types/SizeOf.hs

```diff
@@ -81,157 +96,166 @@ constructorCost numFields = headerCost + (constructorFieldCost numFields)
 
 
 instance (SizeOf v) => SizeOf (Vector v) where
-  sizeOf v = vectorSize
+  sizeOf ver v = vectorSize
     where
       vectorSize =
         ((7 + vectorLength) * wordSize) + sizeOfContents
       vectorLength = fromIntegral (V.length v)
-      sizeOfContents = V.foldl' (\acc pv -> acc + (sizeOf pv)) 0 v
+      sizeOfContents = V.foldl' (\acc pv -> acc + (sizeOf ver pv)) 0 v
 
 instance (SizeOf a) => SizeOf (Set a) where
-  sizeOf s = setSize
+  sizeOf ver s = setSize
     where
       setSize = ((1 + 3 * setLength) * wordSize) + sizeOfSet
       setLength = fromIntegral $ S.size s
-      sizeOfSet = S.foldl' (\acc a -> acc + sizeOf a) 0 s
+      sizeOfSet = S.foldl' (\acc a -> acc + sizeOf ver a) 0 s
 
 instance (SizeOf k, SizeOf v) => SizeOf (M.Map k v) where
-  sizeOf m = mapSize
+  sizeOf ver m = mapSize
     where
       mapSize = (6 * mapLength * wordSize) + sizeOfKeys + sizeOfValues
       mapLength = fromIntegral (M.size m)
-      sizeOfValues = M.foldl' (\acc pv -> acc + (sizeOf pv)) 0 m
-      sizeOfKeys = M.foldlWithKey' (\acc fk _ -> acc + (sizeOf fk)) 0 m
+      sizeOfValues = M.foldl' (\acc pv -> acc + (sizeOf ver pv)) 0 m
+      sizeOfKeys = M.foldlWithKey' (\acc fk _ -> acc + (sizeOf ver fk)) 0 m
 
 instance (SizeOf a, SizeOf b) => SizeOf (a,b) where
-  sizeOf (a,b) = (constructorCost 3) + (sizeOf a) + (sizeOf b)
+  sizeOf ver (a,b) = constructorCost 3 + sizeOf ver a + sizeOf ver b
 
 instance (SizeOf a) => SizeOf (Maybe a) where
-  sizeOf (Just e) = (constructorCost 1) + (sizeOf e)
-  sizeOf Nothing = constructorCost 0
+  sizeOf ver (Just e) = constructorCost 1 + sizeOf ver e
+  sizeOf _ Nothing = constructorCost 0
 
 
 instance (SizeOf a) => SizeOf [a] where
-  sizeOf arr = arrSize
+  sizeOf ver arr = arrSize
     where
       arrSize = ((1 + (3 * arrLength)) * wordSize) + sizeOfContents
       arrLength = fromIntegral (L.length arr)
-      sizeOfContents = L.foldl' (\acc e -> acc + (sizeOf e)) 0 arr
+      sizeOfContents = L.foldl' (\acc e -> acc + (sizeOf ver e)) 0 arr
 
 instance SizeOf BS.ByteString where
-  sizeOf bs = byteStringSize
+  sizeOf _ bs = byteStringSize
     where
       byteStringSize = (9 * wordSize) + byteStringLength
       byteStringLength = fromIntegral (BS.length bs)
 
 instance SizeOf Text where
-  sizeOf t = (6 * wordSize) + (2 * (fromIntegral (T.length t)))
+  sizeOf _ t = (6 * wordSize) + (2 * (fromIntegral (T.length t)))
 
 instance SizeOf Integer where
-  sizeOf i = ceiling ((logBase 100000 (realToFrac i)) :: Double)
+  sizeOf ver i = case ver of
+    SizeOfV0 ->
+      if i <= 0 then 0 else ceiling ((logBase 100000 (realToFrac i)) :: Double)
+    SizeOfV1 ->
+#if !defined(ghcjs_HOST_OS)
+      fromIntegral (max 64 (I# (IntLog.integerLog2# (abs i)) + 1))  `quot` 8
+#else
+      (max 64 (ceiling (logBase @Double 2 (fromIntegral (abs i))) + 1)) `quot` 8
```

Comment:
do we even want to maintain code for GHCJS here? Is it covered by a test somewhere? I'd rather not maintain it.

Reply:
GHCJS will literally not compile `integerLog2#` so I don't see what option we have.

Reply:
Can you see if `SizeOfSpec` can move up to the top of the `tests` stanza in `pact.cabal`? That would result in GHCJS being covered.